### PR TITLE
XIVY-11980 generalize tree data handling

### PR DESCRIPTION
**Pull request by ivy-lgi**

### CODE REVIEW

**Review by ivy-lli — APPROVED** [review]
Looks good 👍

**Comment by ivy-lli** [comment]
some basic unit tests would be great for this utility functions :)

File: packages/new-editor/src/utils/tree.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import type { Row, Table } from '@tanstack/react-table';
+import { addNode, getNode, removeNode } from '../data/data';
+import type { TreeNode, TreePath, addChildToFirstSelectedRowReturnType } from '../types/config';
+import { getFirstSelectedRow, selectRow } from './table';
+
+export const addChildToFirstSelectedRow = <TNode extends TreeNode<TNode>>(
```
Comment (weissreto) [inline packages/new-editor/src/utils/tree.ts:6]: Should these become part of our UI widgets?

Comment (ivy-lli) [reply packages/new-editor/src/utils/tree.ts:6]: Maybe. First time of such a method. 👍 
The inscription view only knows adding in a data table not trees.

**Pull request merged**